### PR TITLE
Fix LifetimeIntrinsicTest issues

### PR DESCRIPTION
**Pull request by tex3d**
- was only run on SM 6.6 supporting devices
- would only run SM 6.6 tests in experimental mode, compiling with -Vd
- wasn't specifying options required to test lifetime fallback paths

### CODE REVIEW

**Review by pow2clk — APPROVED** [review]
Extremely minor quibbles about naming. Looks good!

File: tools/clang/unittests/HLSL/ExecutionTest.cpp [hunk 1]
```diff
@@ -1634,8 +1634,11 @@ void ExecutionTest::RunLifetimeIntrinsicComputeTest(ID3D12Device *pDevice, LPCST
   WaitForSignal(pCommandQueue, FO);
 }
 
-void ExecutionTest::RunLifetimeIntrinsicLibTest(ID3D12Device5 *pDevice, LPCSTR pShader, CComPtr<ID3D12RootSignature>& pRootSignature,
+void ExecutionTest::RunLifetimeIntrinsicLibTest(ID3D12Device *pDevice1, LPCSTR pShader, CComPtr<ID3D12RootSignature>& pRootSignature,
                                                 LPCWSTR pTargetProfile, LPCWSTR *pOptions, int numOptions) {
+  CComPtr<ID3D12Device5> pDevice;
```
[the remark this answers is not in the document]
Comment (tex3d) [reply tools/clang/unittests/HLSL/ExecutionTest.cpp:1639]: Didn't want to change all the uses...

File: tools/clang/unittests/HLSL/ExecutionTest.cpp [hunk 2]
```diff
@@ -1384,10 +1384,10 @@ class ExecutionTest {
   }
 
   void RunRWByteBufferComputeTest(ID3D12Device *pDevice, LPCSTR shader, std::vector<uint32_t> &values);
-  void RunLifetimeIntrinsicTest(ID3D12Device *pDevice, LPCSTR shader, D3D_SHADER_MODEL shaderModel, bool useLibTarget, LPCWSTR *pOptions, int numOptions, std::vector<uint32_t> &values);
+  void RunLifetimeIntrinsicTest(ID3D12Device *pDevice, LPCSTR shader, D3D_SHADER_MODEL shaderModel, bool useLibTarget, llvm::ArrayRef<LPCWSTR> options, std::vector<uint32_t> &values);
   void RunLifetimeIntrinsicComputeTest(ID3D12Device *pDevice, LPCSTR pShader, CComPtr<ID3D12DescriptorHeap>& pUavHeap, CComPtr<ID3D12RootSignature>& pRootSignature,
                                        LPCWSTR pTargetProfile, LPCWSTR *pOptions, int numOptions, std::vector<uint32_t> &values);
-  void RunLifetimeIntrinsicLibTest(ID3D12Device5 *pDevice, LPCSTR pShader, CComPtr<ID3D12RootSignature>& pRootSignature,
+  void RunLifetimeIntrinsicLibTest(ID3D12Device *pDevice1, LPCSTR pShader, CComPtr<ID3D12RootSignature>& pRootSignature,
```
[the remark this answers is not in the document]
Comment (tex3d) [reply tools/clang/unittests/HLSL/ExecutionTest.cpp:1390]: Yeah, basically I didn't want to change all the uses of pDevice internally that used v5 in the method, so had to rename the non-v5 version, and put a 1 there.  But I could have used a different naming for the parameter than 1, especially considering there actually exists `ID3D12Device1`.

[the remark this answers is not in the document]
Comment (pow2clk) [reply tools/clang/unittests/HLSL/ExecutionTest.cpp:1390]: good enough!

**Pull request merged**